### PR TITLE
Fix blocking ResultBuilder in WriteProxy

### PR DESCRIPTION
**Pull request by MarinPostma**
The QueryBuilder was being misused (by me) in the `WriteProxy` code. This fixes it by driving the query builder in on the thread pool.

### CODE REVIEW

**Review by Horusiath — APPROVED** [review]
It does fix issues I had on my branch:

> Turmoil tests for replication were catching tokio runtime panics caused by spawning task on `HranaStream::drop` without awaiting on it's completion while `CursorResponseBuilder` on the server side was blocking on emitting responses.

The only concern I have here is about internals of spawn_blocking. Won't it cause make_request to become much more resource intensive after this change?

File: libsql-server/src/connection/write_proxy.rs [hunk 1]
```diff
@@ -332,7 +338,23 @@ where
...
+                        let cb_context = cb_context.clone();
+                        move || -> Result<bool> {
+                            let mut ctx = cb_context.lock();
+                            let (ref mut response_cb, ref mut arg) = *ctx;
+                            if !response_cb(resp.response.ok_or(Error::PrimaryStreamMisuse)?, arg)?
+                            {
+                                Ok(false)
+                            } else {
+                                Ok(true)
+                            }
```
Comment (psarna) [inline libsql-server/src/connection/write_proxy.rs:351]: that took quite some brainpower for me to parse, with all the negatives (: Isn't all this reducible to just:
```rust
response_cb(resp.response.ok_or(Error::PrimaryStreamMisuse)?, arg)
```
, after you drop the negations and collapse the `Ok(something?)` pattern?

Comment (psarna) [inline libsql-server/src/connection/write_proxy.rs:351]:
```suggestion
response_cb(resp.response.ok_or(Error::PrimaryStreamMisuse)?, arg)
```

**Review by psarna — COMMENTED** [review]
to clarify: all this code is pretty much needed just to be able to run `apply_program_resp_to_builder` in a blocking thread, right? I wish there was some more straightforward way to do that without the extra magic lifetime parameters, but I can't think of one

**Comment by MarinPostma** [comment]
@sarna, I'll probably go over that code again for a refactor when I have more time. Another approach would be to spawn the builder in blocking thread, and use a channel to apply the steps to it, but I just wanted aquick fix, and it looked like more complex chanes.

**Pull request merged**